x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1040,13 +1014,8 @@ private void findTokenFromRefreshToken(String refreshToken, Iterator<TimeValue>
                     logger.debug("Decoded refresh token [{}] with version [{}] is invalid.", unencodedRefreshToken, refreshTokenVersion);
                     listener.onResponse(SearchHits.EMPTY_WITH_TOTAL_HITS);
                 } else {
-                    // TODO Remove this conditional after backporting to 7.x
-                    if (refreshTokenVersion.onOrAfter(VERSION_HASHED_TOKENS)) {
-                        final String hashedRefreshToken = hashTokenString(unencodedRefreshToken);
-                        findTokenFromRefreshToken(hashedRefreshToken, securityTokensIndex, backoff, listener);
-                    } else {
-                        findTokenFromRefreshToken(unencodedRefreshToken, securityTokensIndex, backoff, listener);
-                    }
+                    final String hashedRefreshToken = hashTokenString(unencodedRefreshToken);
```

Comment:
We've done the backport here: https://github.com/elastic/elasticsearch/pull/42220